### ISSUE TITLE
Clean up FileSystemWatcher tests use of temp files

### ISSUE THREAD

**Issue by danmoseley** [dotnet/runtime#68430]
they make various use of TempDirectory and TempFile, eg.,
```c#
            using (var testDirectory = new TempDirectory(GetTestFilePath()))
            using (var dir = new TempDirectory(Path.Combine(testDirectory.Path, "dir")))
            using (var temp = new TempFile(GetTestFilePath()))
```
but all the test classes derive from FileCleanupTestBase. This base class automatically cleans up all files and directories from a test after the test completes, if the test puts them in `TestDirectory` (a property on FileCleanupTestBase) or gets them from `GetTestFilePath()`.

We can probably remove all use of TempDirectory and TempFile in favor of creating directories within `TestDirectory` and creating files with `GetTestFilePath()`.

**Comment by danmoseley** [comment]
@Danyy427 if you are interested.

**Comment by Danyy427** [comment]
@danmoseley Yes, sure. I’ll have to get to it in the morning (3 am. here) though.

**Comment by Danyy427** [comment]
@danmoseley I did not understand what I am supposed to do. Am I meant to remove `TempFile()` and `TempDirectory()` stuff and replace them with things like `File.Create(GetTestFilePath());` or make `GetTestFilePath()` create the files themselves (because as far as I understood it doesn't actually create the files).

**Comment by danmoseley** [comment]
`string dir = Path.Combine(GetTestFilePath(), "dir");` in think would be `string dir = Path.Combine(TestDirectory, "dir");` in your example I think

**Comment by Danyy427** [comment]
@danmoseley So, I will indeed replace `TempFile()` and `TempDirectory()` with `File.Create` and etc. 

Also the original code:

```
using (var testDirectory = new TempDirectory(GetTestFilePath()))
using (var dir = new TempDirectory(Path.Combine(testDirectory.Path, "dir")))
```
suggests that a new directory is opened on top of `GetTestFilePath()`

**Comment by danmoseley** [comment]
File.Create returns a FileStream. I think you can dispose it in the same line of code.

**Comment by Danyy427** [comment]
@danmoseley oh damn, I totally forgot about that FileStream and have been trying to understand which program was still using my files. Thanks

**Comment by Danyy427** [comment]
@danmoseley just to clarify should I use `GetTestFilePath()` for `testDirectory` per the original code, or use `TestDirectory`. It shouldn't matter but what is the correct convention?

**Comment by danmoseley** [comment]
GetTestFilePath() gives you a path inside TestDirectory. If you don't need that extra part, TestDirectory is simpler .

**Comment by Danyy427** [comment]
@danmoseley in the tests I always put the outer part at `TestDirectory` and didn't use `GetTestFilePath` unless needed

**Comment by Danyy427** [comment]
@danmoseley I added the PR

**Comment by danmoseley** [comment]
Just cleanliness - don't do more than you need to.

**Comment by danmoseley** [comment]
Fixed by #68430

@Danyy427  when you open a PR that fixes an issue, you can include a line on the description like "Fixes #68430" and it will code the issue when it merges, even if it's not your issue.

**Comment by Danyy427** [comment]
@danmoseley I will keep that in mind, thank you.
